src/slurm_plugin/clustermgtd.py

```diff
@@ -907,7 +908,69 @@ def _run_clustermgtd():
                 e,
             )
         # Manage cluster
-        cluster_manager.manage_cluster()
+        cluster_manager.manage_cluster(custom_metrics)
+        custom_metrics._has_received = True
+        sleep_remaining_loop_time(config.loop_time, start_time)
+
+
+class ClusterMetrics:
+    def __init__(self):
+        self._has_received = False
```

Comment:
Instead of having an attribute for each metric we could simply have a dict called `metrics`.

Here, we can define an empty dict and in the `manage_cluster` we can put metrics in the dict.

Then in the get_values we just need to cycle the metrics dict and push the metrics we have.
In this way we can get rid of the `_has_received` parameter and we can easily add code somewhere to save new metrics, without the need to change the ClusterMetrics class or the upload method.

Another OO alternative could be to define a Metric class, that could have name and value attribute and in the ClusterMetrics we can manage a list of Metric objects, but for our initial case I think a dict is enough because it also permit to easily update an already existing metric because we'll use the name as dict key.

Reply:
We now have a timestamp and check if it has not been too much time since we last got new metrics. We also have two dictionaries now (metrics and metrics_provider) that enable us to add new metrics in a flexible manner.